### PR TITLE
clay: support labels

### PR DESCRIPTION
**Pull request by ynx0**
This PR finishes off the implementation for using labels in the clay filesystem. Labels are `@tas` which you can use in place of dates or revision numbers.

<details>
<summary>Examples to try</summary>
Make a change to `<ship>/base/gen/hood/label.hoon` after mounting, then run the following

```hoon
|label %base %v2
|label %base %origin, =aeon 1

.^(@t %cx /=base/v2/gen/hood/label/hoon)
.^(@t %cx /=base/origin/gen/hood/label/hoon)
```

</details>

Many thanks to @philipcmonk who worked through this with me.

### CODE REVIEW

File: pkg/arvo/sys/vane/clay.hoon [hunk 1]
```diff
@@ -1445,6 +1445,32 @@
...
+    |=  [bel=@tas aey=(unit aeon)]
+    ^+  ..park
+    =/  yon  ?~(aey let.dom u.aey)
+    =/  yen  (~(get by lab.dom) bel)  :: existing aeon?
+    ::  no existing aeon is bound to this label
+    ::
+    ?~  yen
+      =.  lab.dom  (~(put by lab.dom) bel yon)
+      ..park
+    :: an aeon is bound to this label,
```
Comment (philipcmonk) [inline pkg/arvo/sys/vane/clay.hoon:1460]: By convention (at least in the kernel?) comments in hoon start with two spaces, so they're aligned with runes.

File: pkg/arvo/sys/vane/clay.hoon [hunk 2]
```diff
@@ -1445,6 +1445,32 @@
...
+    ::
+    ?:  =(u.yen yon)
+      ~&  "clay: tried to rebind existing label {<bel>} to equivalent aeon {<yon>}"
+      ..park
+    :: an existing aeon bound to the label
+    :: that is distinct from the requested one.
+    :: rewriting would violate referential transparency
+    ::
+    ~|  %tried-to-rewrite-existing-label
+    ~&  "requested aeon: {<yon>}, existing aeon: {<u.yen>}"
```
Comment (philipcmonk) [inline pkg/arvo/sys/vane/clay.hoon:1471]: Better to put all the info in the stack trace:

```suggestion
    ~|  "requested aeon: {<yon>}, existing aeon: {<u.yen>}"
```

**Review by philipcmonk — CHANGES_REQUESTED** [review]
Looks great!  Just a couple of style nits

**Comment by ohAitch** [comment]
When did this even break QQ
seems like a useful capability!

**Review by philipcmonk — APPROVED** [review]
1.5 years ago with Ford Fusion

Thanks, this looks good!

**Pull request merged**